### DISCUSSION TITLE
Temperature, HTTP-Request

### DISCUSSION THREAD

**Discussion by BernhardSchlegel** [arendst/Tasmota#10573] [Q&A] [answered]
Hi everyone, 

I want to implement (stuff that I think I figured out is denoted by () ) for my Sonoff TH16

- Acquiring the temperature from the DS18B20
- Posting a JSON to a REST-Endpoint
- (Calling it every 10s: `TimeReached(state_10second)` in tasmota.h)
- (implementing a custom setting menue)

I'm very thankfull for any input (sample code, reference to a file from the repo, buzzword that might show the direction) concerning:

1. How to get the temperature? (I tried getting the temperature using `mqtt_data`, `ds18x20_sensor` but that's not accessible)
2. How to create a JSON object out of it? (Can I use e.g. `ResponseAppend_P`)
3. What's the slimmest way of including something for `http.get(target_url, payload)`?

I used to be not too bad with `C`, but I'm a little overwhelmed by the amount of helper function based string concatenations and function pointer callbacks.

Thanks in advance!

**Comment by ascillato** [comment]
Hi, you don't need to program or modify Tasmota for that. It is already supported. And there are plenty of examples in the docs and in old issues. Please do a search.

You can use the Tasmota feature called RULES and the command WEBSEND.

Please, read the docs. Thanks.

**Comment by barbudor** [comment, marked as answer]
You can implement this using Rules, Websend and setting teleperiod to 10 seconds

Something like 
`on tele-DS18B20#Temperature do Websend [serverip] /rest/api/endpoint?%value%`
should do it

**Reply by BernhardSchlegel** [reply]
Is it possible to get the returned value from `Websend`? Say my REST enpoint returns a simple 1 / 0 (not even JSON for that matter). As [I understood it](https://tasmota.github.io/docs/Commands/) that's not possible? I would need something like "`Webget`" - would that be a PR that is likely to be accepted?

**Reply by barbudor** [reply]
Not by default when using rules
If you use [Scripting-Language](https://tasmota.github.io/docs/Scripting-Language/) instead, it is possible to get the response from a WebSend.
To use Scripting, you need either to recompile (see top of the doc page) or go to github.com/tasmota/binaries for a "unsupported" pre-compiled version of tasmota with scripting

**Comment by BernhardSchlegel** [comment]
Can I get you guys a beer? https://paypal.me/tasmota ?
